examples/Darcy/darcy.jl

```diff
@@ -0,0 +1,179 @@
+include("GModel.jl")
+
+# Then link some outputs for figures and plotting
+fig_save_directory = joinpath(@__DIR__, "output")
+data_save_directory = joinpath(@__DIR__, "output")
+if !isdir(fig_save_directory)
+    mkdir(fig_save_directory)
+end
+if !isdir(data_save_directory)
+    mkdir(data_save_directory)
+end# TOML interface for fitting parameters of a sinusoid
+
+PLOT_FLAG = true
+if PLOT_FLAG
+    using Plots
+    @info "Plotting enabled, this will reduce code performance. Figures stored in $fig_save_directory"
+end
+
+# Set a random seed.
+seed = 100234
+rng = Random.MersenneTwister(seed)
+
+# Define the spatial domain and discretization 
+dim = 2
+N, L = 80, 1.0
+pts_per_dim = LinRange(0, L, N)
+obs_ΔN = 10
+
+# To provide a simple test case, we assume that the true function parameter is a particular sample from the function space we set up to define our prior. More precisely we choose a value of the truth that doesnt have a vanishingly small probability under the prior defined by a probability distribution over functions; here taken as a family of Gaussian Random Fields (GRF). The function distribution is characterized by a covariance function - here a Matern kernel which assumes a level of smoothness over the samples from the distribution. We define an appropriate expansion of this distribution, here based on the Karhunen-Loeve expansion (similar to an eigenvalue-eigenfunction expansion) that is truncated to a finite number of terms, known as the degrees of freedom (`dofs`). The `dofs` define the effective dimension of the learning problem, decoupled from the spatial discretization. Explicitly, larger `dofs` may be required to represent multiscale functions, but come at an increased dimension of the parameter space and therefore a typical increase in cost and difficulty of the learning problem.
+
+smoothness = 1.0
+corr_length = 0.25
+dofs = 50
+
+grf = GRF.GaussianRandomField(
+    GRF.CovarianceFunction(dim, GRF.Matern(smoothness, corr_length)),
+    GRF.KarhunenLoeve(dofs),
+    pts_per_dim,
+    pts_per_dim,
+)
+
+# We define a wrapper around the GRF, and as the permeability field must be positive we introduce a domain constraint into the function distribution. Henceforth, the GRF is interfaced in the same manner as any other parameter distribution with regards to interface.
+pkg = GRFJL()
+distribution = GaussianRandomFieldInterface(grf, pkg) # our wrapper from EKP
+domain_constraint = bounded_below(0) # make κ positive
+pd = ParameterDistribution(Dict("distribution" => distribution, "name" => "kappa", "constraint" => domain_constraint)) # the fully constrained parameter distribution
+
+# Now we have a function distribution, we sample a reasonably high-probability value from this distribution as a true value (here all degrees of freedom set with `u_{\mathrm{true}} = -0.5`). We use the EKP transform function to build the corresponding instance of the ``\kappa_{\mathrm{true}}``.
+u_true = -1.5 * ones(dofs, 1) # the truth parameter
```

Comment:
I would find it useful to justify the definition of u_true = -1.5. I recall from our meeting that it is just an arbitrary value, but the docs do not indicate that it is arbitrary, which can be misleading.

Reply:
I have done so in darcy.md